atc/atccmd/command.go

```diff
@@ -179,6 +180,7 @@ type RunCommand struct {
 	BuildTrackerInterval time.Duration `long:"build-tracker-interval" default:"10s" description:"Interval on which to run build tracking."`
 
 	TelemetryOptIn bool `long:"telemetry-opt-in" hidden:"true" description:"Enable anonymous concourse version reporting."`
+	ZstdOptIn      bool `long:"zstd-opt-in" hidden:"true" description:"Enable zstd compression."`
```

Comment:
Was there a specific reason to make it `hidden`? Also, I think we should go with `enable-*` instead given that we've been doing that for all of the other "opt-in" features:

```console
$ concourse web --help 2>&1 | grep enable
          --enable-global-resources                                                                             Enable equivalent resources
          --enable-build-auditing                                                                               Enable auditing for all api
          --enable-container-auditing                                                                           Enable auditing for all api
          --enable-job-auditing                                                                                 Enable auditing for all api
          --enable-pipeline-auditing                                                                            Enable auditing for all api
          --enable-resource-auditing                                                                            Enable auditing for all api
          --enable-system-auditing                                                                              Enable auditing for all api
          --enable-team-auditing                                                                                Enable auditing for all api
          --enable-worker-auditing                                                                              Enable auditing for all api
          --enable-volume-auditing                                                                              Enable auditing for all api
          --enable-redact-secrets                                                                               Enable redacting secrets in
          --enable-lets-encrypt                                                                                 Automatically configure TLS
          --secret-cache-enabled                                                                                Enable in-memory cache for
          --secret-cache-duration=                                                                              If the cache is enabled,
          --secret-cache-duration-notfound=                                                                     If the cache is enabled,
          --secret-cache-purge-interval=                                                                        If the cache is enabled,
```

Reply:
nothing special here just follow the flag that has "OptIn" in it.
One reason I hesitated to make it a "enable" flag is we know there is still memory issue around it, which make me feel like we should not encourage user to opt in zstd until we figure it all out. I could make it not hidder for sure.

Reply:
The way around is we make it a `enable` like feature flag and dogfooting in our own CI so we could observe and capture potential issues that mentioned in https://github.com/concourse/concourse/pull/5183

WDYT?

Reply:
makes sense! thaanks!